packages/compiler/test/core/numeric.test.ts

```diff
@@ -39,6 +39,11 @@ describe("parsing", () => {
     it("simple decimal", () => {
       expectNumericData("123.456", 123456n, 3);
     });
+
+    it("decimal with trailing zeros", () => {
```

Comment:
I agree it's surprising this failed (you said `1.0` and `10.0` passed).  I recommend adding:

1. More tests with varying numbers of zeros before and after the decimal (e.g. `1.00000`, `10000000.000000`)
2. Maybe a basic "fuzz test" where you pick a bunch of random numbers, serialize them (with varying formats and degrees of precision), and verify they round-trip correctly.

Reply:
both of those worked because I was not comparing the decimal index correctly so in both cases it was looking one more number to the left so it wasn't 0 and didn't get stuck in those loops

Reply:
but yeah i'll add more test